### PR TITLE
Monitoring

### PR DESCRIPTION
**Pull request by alessio-creo**
Needs #42

### CODE REVIEW

File: src/main/60-eks.tf [hunk 1]
```diff
@@ -66,6 +66,12 @@ module "eks" {
         computeType = "Fargate"
       })
     }
+    adot = {
+      addon_version               = var.kubernetes_addons_versions.adot
+      resolve_conflicts_on_create = "OVERWRITE"
+      resolve_conflicts_on_update = "PRESERVE"
+
+    }
```
Comment (micdes-pagopa) [inline src/main/60-eks.tf:74]: Are we using the addon? It looks like we are creating ADOT with yaml manifests in `k8s/assets/adot_collector`

File: src/main/k8s/assets/adot_collector/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    metadata:
+      labels:
+        app: aws-adot
+        component: adot-collector
+    spec:
+      serviceAccountName: adot-collector
+      securityContext:
+        fsGroup: 65534
+      containers:
+        - image: public.ecr.aws/aws-observability/aws-otel-collector:latest
```
Comment (micdes-pagopa) [inline src/main/k8s/assets/adot_collector/deployment.yaml:24]: The image version must be pinned! If they release a breaking change we could lose metrics. Use a TF variable and parametrize

File: src/main/k8s/assets/metrics_server/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        k8s-app: metrics-server
+    spec:
+      containers:
+      - args:
+        - --cert-dir=/tmp
+        - --secure-port=4443
+        - --kubelet-preferred-address-types=InternalIP,ExternalIP,Hostname
+        - --kubelet-use-node-status-port
+        - --metric-resolution=15s
+        image: registry.k8s.io/metrics-server/metrics-server:v0.6.3
```
Comment (micdes-pagopa) [inline src/main/k8s/assets/metrics_server/deployment.yaml:27]: Image version should be passed as a TF variable

File: src/main/k8s/assets/xray_daemon/svc.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  namespace: ${namespace}
+spec:
+  ports:
+  - name: "http"
+    port: 80
+    protocol: UDP
+    targetPort: 2000
+  selector:
+    app: xray-daemon
+  type: NodePort
```
Comment (micdes-pagopa) [inline src/main/k8s/assets/xray_daemon/svc.yaml:16]: Should be `ClusterIP`

**Review by micdes-pagopa — CHANGES_REQUESTED** [review]

File: src/main/60-eks.tf (hunk 1 above)
Comment (alessio-creo) [reply src/main/60-eks.tf:74]: This is the operator installation. In the k8s section is the collector installation

Comment (micdes-pagopa) [reply src/main/60-eks.tf:74]: Do you need both? We usually scrape cadvisor metrics and kube state metrics using only `aws-otel-collector`. If you feel that both are needed in this case that's okay, as long as metrics are not duplicated

File: src/main/k8s/assets/metrics_server/deployment.yaml (hunk 3 above)
Comment (alessio-creo) [reply src/main/k8s/assets/metrics_server/deployment.yaml:27]: done

File: src/main/k8s/assets/adot_collector/deployment.yaml (hunk 2 above)
Comment (alessio-creo) [reply src/main/k8s/assets/adot_collector/deployment.yaml:24]: done

File: src/main/k8s/assets/xray_daemon/svc.yaml (hunk 4 above)
Comment (alessio-creo) [reply src/main/k8s/assets/xray_daemon/svc.yaml:16]: done

**Pull request merged**